Description: Skip package tests that have been known to fail on
 various architectures.
Author: Doug Torrance <dtorrance@debian.org>
Bug: https://github.com/Macaulay2/M2/issues/1539
Bug: https://github.com/Macaulay2/M2/issues/1579
Bug: https://github.com/Macaulay2/M2/issues/1742
Bug: https://github.com/Macaulay2/M2/issues/1903
Last-Update: 2021-12-16

--- a/M2/Macaulay2/m2/testing.m2
+++ b/M2/Macaulay2/m2/testing.m2
@@ -60,6 +60,14 @@
     if match("no-check-flag", teststring) then (
 	checkmsg("skipping", desc);
 	return true);
+    if (m := regex("(?<=no-check-architecture:)[^(\\n]*", teststring)) =!= null
+    then (
+	badarchs := apply(separate(", ", substring(m#0, teststring)),
+	    s -> replace("^\\s*|\\s*$", "", s));
+	if member(version#"architecture", badarchs) then (
+	    checkmsg("skipping", desc);
+	    return true));
+
     -- TODO: remove this when capture uses ArgQ
     if usermode === not noinitfile then
     -- try capturing in the same process
--- a/M2/Macaulay2/packages/SumsOfSquares.m2
+++ b/M2/Macaulay2/packages/SumsOfSquares.m2
@@ -1437,6 +1437,7 @@
 
 --11
 TEST /// --lowerBound
+-- no-check-architecture: aarch64, i686, powerpc64le, s390x (#1579)
     debug needsPackage "SumsOfSquares"
     results := checkLowerBound("CSDP")
     assert all(results,t->t=!=false);
--- a/M2/Macaulay2/packages/SpecialFanoFourfolds.m2
+++ b/M2/Macaulay2/packages/SpecialFanoFourfolds.m2
@@ -2887,6 +2887,7 @@
 ///
 
 TEST /// -- Test 2 (2/2) -- GM fourfolds from strings: describe, discriminant, parameterCount, toGrass
+-- no-check-architecture: arm (#1539)
 strIn := {"cubic scroll", "quintic del Pezzo surface", "general GM 4-fold of discriminant 20"};
 strOut := "Special Gushel-Mukai fourfold of discriminant 12
 containing a surface in PP^8 of degree 3 and sectional genus 0
@@ -2919,6 +2920,7 @@
 ///
 
 TEST /// -- Test 3 -- 21 examples from GMtables
+-- no-check-flag #1539, #1742
 X = for i from 1 to 21 list (
    A = GMtables(i,ZZ/65521);
    time specialGushelMukaiFourfold((rationalMap(ideal A_0,Dominant=>2)) ideal A_1,InputCheck=>0)
@@ -3021,6 +3023,7 @@
 ///
 
 TEST /// -- Test 14 -- gluing scrolls along curves
+-- no-check-flag #1742
 debug SpecialFanoFourfolds
 S = surface({3,4,0,0},ambient=>6);
 for a in {(1,0),(2,0),(3,0),(4,0),(5,0),(5,1)} do (
@@ -3045,6 +3048,7 @@
 ///
 
 TEST /// -- Test 15
+-- no-check-architecture: x86_64 (#1742)
 debug SpecialFanoFourfolds
 L = takeGMsfromSurfaceInP6(surface({3,1,1,0},ambient=>6),InputCheck=>0,"Gluing"=>"cubic scroll",Degrees=>hashTable{1=>(1,1),2=>(19,infinity),3=>(0,0)});
 X = first L;
--- a/M2/Macaulay2/packages/MultiGradedRationalMap.m2
+++ b/M2/Macaulay2/packages/MultiGradedRationalMap.m2
@@ -1272,6 +1272,7 @@
 
 
 TEST ///
+-- no-check-architecture: x86_64, s390x (#1742)
      R = QQ[x,y,z]
      A = matrix{ {x, x^5 + y^5},
                  {-y, y^5 + z*x^2*y^2},
--- a/M2/Macaulay2/packages/MultiprojectiveVarieties.m2
+++ b/M2/Macaulay2/packages/MultiprojectiveVarieties.m2
@@ -3862,6 +3862,7 @@
 ///
 
 TEST /// -- parametrizations 2
+-- no-check-architecture: x86_64 (#1742)
 checkInverseParametrization = X -> (
     f := parametrize X;
     <<f<<endl;
--- a/M2/Macaulay2/packages/AssociativeAlgebras/tests.m2
+++ b/M2/Macaulay2/packages/AssociativeAlgebras/tests.m2
@@ -1134,6 +1134,7 @@
 ///
 
 TEST ///
+-- no-check-flag #1903
 -- testing kernels
 -*
   restart
--- a/M2/Macaulay2/packages/SimplicialDecomposability.m2
+++ b/M2/Macaulay2/packages/SimplicialDecomposability.m2
@@ -1025,6 +1025,7 @@
 
 -- Tests isDecomposable: see Examples V6F10-{1,6,7} in [MT].
 TEST ///
+-- no-check-flag #1742
 R = QQ[a..f];
 S1 = simplicialComplex {a*b*c, a*b*d, a*b*f, a*c*d, a*c*e, b*d*e, b*e*f, c*d*f, c*e*f, d*e*f};
 S6 = simplicialComplex {a*b*c, a*b*d, a*b*e, a*c*d, a*c*f, b*d*e, b*e*f, c*d*f, c*e*f, d*e*f};
